<a href="https://colab.research.google.com/github/evroth/gsb545repo/blob/main/personal_finance.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Personal Finance Activity

## Question 1
*What were your reasons for pursuing this MSBA program here at Cal Poly? Did you come straight from another academic program (Bachelors, Masters, etc.) or were you working before this? What program(s) or job did you come from?*

I chose to pursue this MSBA program for many reasons. One is I wanted to get a leg up in the job market, so having a stronger technical background and a masters degree would help, especially down the line in a job. I came straight from my undergrad in econ here, which I did in two years and two quarters, so that leads to reason two why I wanted to do the program. I wanted my fourth year in college here. I wasn't ready to leave. Also reason number 3 is that this program is much more affordable than ucla or similar program out of state, thought it is still expensive tho for half of the classes being the same as undergrad- maybe thats just how the system works.

## Question 2
*What job or career do you hope to pursue after graduating from this MSBA program? Do you have particular industries or companies in mind? If so, what are they? Do you have particular places in mind that you'd like to live or work? Why are you interested in these companies, etc.?*

At the begininng of the year I wanted to go into some business analyst or data analyst role. I wanted to go into tech and/or a large company to learn at first. I was not settled on a place, I was looking out of state, but if I stayed in the state I would have looked at SD or maybe SF i it paid right. I just wanted a new experience and I big city could be fun after living in slo for 4 years. In October I took a job with AT&T so I guess I kind of know what I'll be doing which has been nice. I am doing a data analyst role in a two ish year rotational which should be fun and let me explore and learn a lot. I'll be moving to their HQ in Dallas which is quite a shift, but I know quite a few people there and they all enjoy it. I have heard it is a decent company to work for and the work life balance is not bad, which is big for me. 

## Finance Stuff


*What is your estimate of the cost-of-living in your desired place to live, per month? You should justify this estimate by citing online sources or other relevant information. You should also break this down into component costs (e.g. housing, utilities, car, gas, internet, food, entertainment, etc.) as much as you can.*

Some of the numbers taken from https://www.numbeo.com/cost-of-living/in/Dallas, https://www.upnest.com/1/post/cost-of-living-dallas/#:~:text=Food,%24775%20per%20month%20on%20food.



In [ ]:
# Costs

# From ATT website minus any fees, add discount, includes family's phones
home_fiber = 27.5
phone_bill = 100

# Food
cheap_dining = 20
nice_dining = 35
groceries = 300
protein_powder = 35

# Dog's Stuff
dog_food = 70
dog_other = 30

# Transportation
car_insurance = 200
gas = 3.15
oil_change = 70

# Housing
rent = 1550
utilities = 200
pet_rent = 50
covered_parking = 50

# Misc Costs
gym = 30 #discounted
sports_game = 75
fun = 300
bills_other = 100
rand_healthcare = 50
vacation_fund = 200

In [ ]:
# Random Variables
amount_nice_dining = 5
amount_cheap_dining = 10
dog_food_lasts = 1.5 #months
amount_sports_games = 1
miles_to_work = 25
office_days = 14
other_driving_miles = 10

In [ ]:
# Monthly Costs

print("""Monthly Costs:
""")

# Phone/Internet
phone_internet_monthly = home_fiber + phone_bill
print("Phone/Internet: " + str(phone_internet_monthly))

# Food
food_monthly = groceries + (nice_dining*amount_nice_dining) + (cheap_dining*amount_cheap_dining) + protein_powder
print("Food: " + str(food_monthly))

# Dog
dog_monthly = round(((dog_food/dog_food_lasts) + dog_other),2)
print("Dog: "  + str(dog_monthly))

# Transportation
truck_mpg = 16
gas_monthly = (((miles_to_work*2*office_days)/truck_mpg)*gas) + (((other_driving_miles*30)/truck_mpg)*gas)
other_truck_costs = (oil_change/6) + car_insurance
transport_monthly = round((gas_monthly + other_truck_costs),2)
print("Transportation: " + str(transport_monthly))

# Housing
housing_monthly = rent + utilities + pet_rent + covered_parking
print("Housing/Utilities: " + str(housing_monthly))

# Misc Costs
misc_monthly = gym + (sports_game*amount_sports_games) + fun + bills_other + rand_healthcare + vacation_fund
print("Misc Costs: " + str(misc_monthly))


Monthly Costs:

Phone/Internet: 127.5
Food: 710
Dog: 76.67
Transportation: 408.54
Housing/Utilities: 1850
Misc Costs: 755


In [ ]:
# Total Monthly Cost of Living

cost_of_living = phone_internet_monthly + food_monthly + dog_monthly + transport_monthly + housing_monthly + misc_monthly

print("""Total Monthly Cost of Living:
""" + str(cost_of_living))

Total Monthly Cost of Living:
3927.71


*What is the annual salary you need to afford the costs you outlined above? This should, in essence, be the minimum salary you need to afford the life you could live happily (you could probably survive on less, but let's not get too detailed here). Be careful to consider whether you're taking taxes into account! We don't need to get too detailed here either, but you should try to estimate what your actual take-home pay is versus the gross salary that you might be presented with in the offer from your employer.*

*Note this is minimum salary for cost of living, not retirement, savings, etc...*

*Tax rate from https://smartasset.com/taxes/texas-tax-calculator#m45sM8kNTq*

In [ ]:
# Annual Salary Minimum

post_tax_min = cost_of_living*12

tax_rate = .23 # approx avg tax rate at actual income level, would be lower in this situation

min_annual_salary = round((post_tax_min/(1-tax_rate)),2)

print("""Minimum Annual Salary Minimum:
""" + str(min_annual_salary))

Minimum Annual Salary Minimum:
61211.06


*What is the annual salary you WANT? I imagine this will be higher than what was previously mentioned. What do you want the additional income for?*

In [ ]:
# Other Wants 

# annual savings
roth_ira = 6500
savings = 20000
job_401k = 16600

# fun things
new_phone = 0
new_car = 0

function below to play around with company match stuff

In [ ]:
def calc_401k_savings(salary: float, percentage: float) -> float:

    if percentage < .06:
        raise ValueError(f"Hey dumbass you better at least get the free money from your employeer!")

    match_first_3pct = salary*.03*(4/3)
    match_second_3pct = salary*.03

    base_match = (match_first_3pct + match_second_3pct + (salary*percentage))
 
    return base_match


In [ ]:
calc_401k_savings(115000, .075)

16675.0

Actual Salary I want:

for simplicity as of now I am not doing the tax deduction for 401k contribution, but you would see a slight savings

In [ ]:
total_monthly_costs = round((cost_of_living + (roth_ira/12) + (savings/12) + (job_401k/12) + new_phone + new_car),2)

print("""Total Monthly Costs:
""" + str(total_monthly_costs))

Total Monthly Costs:
7519.38


In [ ]:
# Wanted Annual Salary

annual_post_tax = total_monthly_costs*12

wanted_annual_salary = round((annual_post_tax/(1-tax_rate)),2)

print("""Wanted Annual Salary:
""" + str(wanted_annual_salary))

Wanted Annual Salary:
117185.14


Not too bad

*Finally, when do you want to retire and with how much money? This is where the bulk of your computations for this assignment will be, and you may need to do some research into how these things are computed. Your job may have retirement funding built into it; it may not. Additionally, there are some calculators online that do some of these types of calculations for you. I'd like you to (re-)do these calculations as much as possible on your own in Python.*

I want to retire in my 40s or at least be able to pivot to a lower paying job where I can do something good to give back or start my own business. I built out below to just get a feel if the amount I would like to save/invest is enough to get to retirement and then the rest of my life.

Given the assumptions: I live to 95, I retire at 60, I want the equivalent of 3750 a month in today money each month in retirement, and there is an annual inflation rate of .03... By the age 60 I should have about 3.5 million in then dollars saved to support that. Using nerd wallet as reference. This probably isn't too exact, there are many other factors.

In [ ]:
# Retirement stuff

current_age = 22
retirement_age = 47
distribution_age = 60
retirement_savings = 1500000 # in today money
invested_return = .07
inflation = .03


In [ ]:
def retirement_calculator(retirement_age, distribution_age):
    if retirement_age <= distribution_age:
        years_until_distribution = distribution_age - retirement_age

        fv_401k = round((job_401k*(((1+invested_return)**(retirement_age-current_age)-1)/(invested_return))),2)
        fv_roth = round(((roth_ira*((1+invested_return)**(retirement_age-current_age)-1)/(invested_return))),2)

        retire_401k_amnt = round((fv_401k*((1+invested_return)**(years_until_distribution))),2)
        retire_roth_amnt = round((fv_roth*((1+invested_return)**(years_until_distribution))),2)

        pv_401k = round((retire_401k_amnt/((1+inflation)**(distribution_age-current_age))),2)
        pv_roth = round((retire_roth_amnt/((1+inflation)**(distribution_age-current_age))),2)

        fv_savings = round(((savings*((1+invested_return)**(retirement_age-current_age)-1)/(invested_return))),2) # saved til retirement
        pv_savings = round((fv_savings/((1+inflation)**(retirement_age-current_age))),2) # value today of what is saved

        print(f"""401k Amount at {retirement_age}: 
""" + str(fv_401k) + """ 
""")
        print(f"""401k Amount at Distribution age {distribution_age}: 
""" + str(retire_401k_amnt) + """ 
""")
        print("""What 401k is worth in today $: 
""" + str(pv_401k) + """ 
""")
        print(f"""Roth Amount at {retirement_age}: 
""" + str(fv_roth) + """ 
""")
        print(f"""Roth Amount at Distribution age {distribution_age}: 
""" + str(retire_roth_amnt) + """ 
""")
        print("""What Roth is worth in today $: 
""" + str(pv_roth) + """ 
""")
        print(f"""Saved Amount at {retirement_age}: 
""" + str(fv_savings) + """ 
""")
        print(f"""Savings value in today $ at age retired: 
""" + str(pv_savings) + """ 
""")
        
    else:

        return ValueError(f"You better retire before 60")




In [ ]:
def retirement_calculator_neat(retirement_age, distribution_age):
    if retirement_age <= distribution_age:
        years_until_distribution = distribution_age - retirement_age

        fv_401k = round((job_401k*(((1+invested_return)**(retirement_age-current_age)-1)/(invested_return))),2)
        fv_roth = round(((roth_ira*((1+invested_return)**(retirement_age-current_age)-1)/(invested_return))),2)

        retire_401k_amnt = round((fv_401k*((1+invested_return)**(years_until_distribution))),2)
        retire_roth_amnt = round((fv_roth*((1+invested_return)**(years_until_distribution))),2)

        pv_401k = round((retire_401k_amnt/((1+inflation)**(distribution_age-current_age))),2)
        pv_roth = round((retire_roth_amnt/((1+inflation)**(distribution_age-current_age))),2)

        fv_savings = round(((savings*((1+invested_return)**(retirement_age-current_age)-1)/(invested_return))),2) # saved til retirement
        pv_savings = round((fv_savings/((1+inflation)**(retirement_age-current_age))),2) # value today of what is saved
        
        results = {
            'fv_401k': fv_401k,
            'retire_401k_amnt': retire_401k_amnt,
            'pv_401k': pv_401k,
            'fv_roth': fv_roth,
            'retire_roth_amnt': retire_roth_amnt,
            'pv_roth': pv_roth,
            'fv_savings': fv_savings,
            'pv_savings': pv_savings,
        }
        
        return results
        
    else:

        return ValueError(f"You better retire before 60")


In [ ]:
retirement_calculator(retirement_age,distribution_age)

401k Amount at 47: 
1049934.03 

401k Amount at Distribution age 60: 
2530178.27 

What 401k is worth in today $: 
822880.14 

Roth Amount at 47: 
411118.75 

Roth Amount at Distribution age 60: 
990732.46 

What Roth is worth in today $: 
322212.11 

Saved Amount at 47: 
1264980.75 

Savings value in today $ at age retired: 
604161.85 



In [ ]:
retirement_calculator_neat(retirement_age,distribution_age)

{'fv_401k': 1049934.03,
 'retire_401k_amnt': 2530178.27,
 'pv_401k': 822880.14,
 'fv_roth': 411118.75,
 'retire_roth_amnt': 990732.46,
 'pv_roth': 322212.11,
 'fv_savings': 1264980.75,
 'pv_savings': 604161.85}

In [ ]:
results = retirement_calculator_neat(retirement_age, distribution_age)
total_value = results['retire_401k_amnt'] + results['retire_roth_amnt']
print(total_value)

3520910.73


This will probably work unless they get rid of social security...

Working backwards with this informatio, the rough salary I need to retire when I want with the amount I specified is $117,185.14

Some additional citations:

https://www.wallstreetprep.com/knowledge/time-value-of-money/

https://www.investopedia.com/retirement/calculating-present-and-future-value-of-annuities/

https://www.nerdwallet.com/investing/retirement-calculator
